readme.md

```diff
@@ -15,834 +14,1113 @@
-            evt.editor.doSomething();
-        } );
+	// do something
+}
+```
 
-        // Good too:
-        var that = this;
-        editor.on( 'setData', function( evt ) {
-            editor.setCurrentSelection( that );
-        } );
-    }
+### 2.8.2. Objects minify poorly, closures very well
 
-    function someCallback() {
-        // Should refers to Selection, not Editor.
-        this;
-    }
+Avoid excessing usage of objects, when functional programming may be used as well. Properties and method names unlike private variables and function names cannot be shortened by minifier.
+You can play with [online Google Closure Compiler](http://closure-compiler.appspot.com/home) to test different approaches. Make sure to check the gzipped size which matters the most.
 
-    ```
 
-    C. Use `thisArg`
+## 2.9. Native & Host Objects
 
-    Several prototype methods of ES 5.1 built-ins come with a special `thisArg` signature, which should be used whenever possible. CKEditor follows this practice too, so for example [`eventTarget.on`](http://docs.ckeditor.com/#!/api/CKEDITOR.event-method-on) also accepts context as an argument.
+The basic principle here is:
 
-    ```javascript
+	Don't do stupid shit and everything will be ok.
 
-    // 6.C.1
+To reinforce this concept, please watch the following presentation:
+	
+[“Everything is Permitted: Extending Built-ins” by Andrew Dupont (JSConf2011, Portland, Oregon)](https://www.youtube.com/watch?v=xL3xCO7CLNM)
 
-    var obj = { f: 'foo', b: 'bar', q: 'qux' };
+**But** CKEditor is a widget - an application inside someone's system &ndash; so... do not touch anything except
+`CKEDITOR.*` namespace.
 
-    Object.keys( obj ).forEach( function( key ) {
 
-      // |this| now refers to `obj`
+## 2.10. Comments
+- Single line above the code that is subject (except for long expressions like conditions).
+- Comment is a sentence - start it with capital leter, end with period.
+- Use multiline comments only for API documentation.
+- Check the [JSDuck documentation guide](http://dev.ckeditor.com/wiki/CodeDocumentation).
+- You can use JSDuck documentation style for private functions, but then use single line comments.
+- Justify end line comments with tabs:
 
-      console.log( this[ key ] );
+```javascript
+if ( some != really.complex &&      // Comment line 1.
+	condition || with &&            // Comment line 2.
+	( multiple == !lines ) )        // Comment line 3.
 
-    }, obj ); // <-- the last arg is `thisArg`
+// Of course it's better to avoid this spaghetti at all.
+```
 
-    // Prints...
+# 3. Tests
 
-    // 'foo'
-    // 'bar'
-    // 'qux'
+## 3.1. Manual tests
 
-    ```
+Manual tests reproduction steps should be structured in one, ubiquitous way, to unify test files, so they are easy to read and modify. Manual test file should consist only of a single test suite. Also, avoid putting many test cases in the same file, extract common logic instead using helper methods or template files.
```

Comment:
I'd be careful with the rule of one manual test per file. It can cause the creation of tons of files.

Reply:
Or it might favours the long test wrapped as a single one.

Reply:
But improves the testing phase. You can still create template file, but I'm for discussing this point.